src/Repository/ImageRepository.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Image;
+use App\Entity\User;
+use App\Service\UserAvatarManager;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * @method Image|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Image|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Image[]    findAll()
+ * @method Image[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class ImageRepository extends ServiceEntityRepository
+{
+    /**
+     * @var UserRepository
+     */
+    private $userRepository;
+    /**
+     * @var UserAvatarManager
+     */
+    private $directoryManager;
+
+    public function __construct(
+        RegistryInterface $registry,
+        UserRepository $userRepository,
+        UserAvatarManager $directoryManager
+    ) {
+        parent::__construct($registry, Image::class);
+        $this->userRepository = $userRepository;
+        $this->directoryManager = $directoryManager;
+    }
+
+    /**
+     * @param UploadedFile $uploadedFile
+     * @param User $user
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function uploadUserAvatar(
+        UploadedFile $uploadedFile,
+        User $user
+    ): void {
+        $image = new Image();
+        $filename = $user->getId() . '.' . $uploadedFile->guessExtension();
+
+        $this->directoryManager->saveAvatarInDirectory($uploadedFile, $user);
```

Comment:
The repository is responsible only for interacting with the database. Building the `$image` should happen in other place.